docs/operations/versioning_policy.md

```diff
@@ -0,0 +1,127 @@
+Typically for Kubernetes versions, the latest Kubernetes patch versions of the actual (if not still in _preview_) and the last 3 minor Kubernetes versions are maintained by the community. An operator could define these versions as being _supported_ (e.g. v1.16.1, v1.15.4, v1.14.9 and v1.13.12). 
+For machine images there is usually one supported version per image (e.g coreos).
+
+- **deprecated:** A 'Deprecated' version is a version that approaches the end of its lifecycle and can contain issues which are probably resolved in a supported version. 
+New Shoots should not use this version any more. 
+Existing Shoots will be update to a newer version if configured with AutoUpdate for Kubernetes or Machine image versions.
+Deprecated versions will eventually expire (i.e., removed).
+
+- **expired:** An 'Expired' versions has an expiration date in the past. 
+ New clusters with that version cannot be created and existing clusters are forcefully migrated in their maintenance time to a higher version.
+ 
+ ## Forceful migration of expired versions
+ 
+ If a Shoot is running a version after its expiration date has passed, it will be forcefully migrated during it maintenance time.
+ This happens **even if the owner has opted out of automatic cluster updates!**
+
+For **Machine images**, the Shoots worker pools will be updated to the latest 'non-preview' version of the pools respective image.
+
+For **Kubernetes versions**, the forceful update picks the latest 'non-preview' patch version of the current minor version. 
+
+ If the cluster is already on the latest patch version and the latest patch version is also expired, 
+ it will continue with the latest patch version of the **next consecutive minor Kubernetes version**, so **it will result in an 
+ update of a minor Kubernetes version!** 
+ 
+ If that's expired as well, the update process repeats until a non-expired Kubernetes version is reached.
+  
+ **Depending on the circumstances described above, it can happen that the cluster receives multiple consecutive minor Kubernetes version updates!**.
+ As a Kubernetes version update cannot skip a minor version, the CloudProfile has to be maintained properly.
+
+## Duties of the Gardener Operator
+
+Even though we strive for high automation, we think that it leads to safer operations 
+having a human operator in charge of the following operations: 
+
+- Promotion of a `preview` version to `supported`
+- Deletion of versions (cannot be in use by any shoot any more and need to have an expiration date in the past)
+- Setting a `expiration date` on the latest patch version of an 'unsupported' Kubernetes minor version
+    - After the deprecation date passed, it will force a kubernetes minor version update during the next maintenance time window of the Shoot.
+
+## Version Requirements
+
+The Gardener API server enforces the following:
+
+**Deletion of Kubernetes Versions** 
+- Kubernetes versions that are still in use by a Shoot cannot be deleted
```

Comment:
Previously removing a patch version from the CloudProfile would update the Shoots running this version to the latest version of that particular minor. If they are already running the latest version nothing happens.

Now we would also have to set this version as being deprecated to use the auto update to the latest patch version of the minor. To force a minor version update we would have to set an expiration date. If we want to force the update in the next maintenance cycle, we would have to set the expiration date to the past.

Reply:
> Previously removing a patch version from the CloudProfile would update the Shoots running this version to the latest version of that particular minor.

This is not true, if `.spec.maintenance.autoUpdate.kubernetesVersion=false`. See gardener/gardener#1423.

So with this PR, if we introduce a new Kubernetes patch version (latest for the minor) to a CloudProfile and decide to remove it after some time, we have to set its `expirationDate` in the past which will trigger minor version update for the Shoots that are using this patch version?

Reply:
We have to make sure that no Shoot runs that version anymore. 
We can either do that by
-  introducing a newer patch version and let Shoots auto update or set expiration date and let Shoots auto update.
- letting the latest version expire and therefore forcing minor version upgrades
Only if no Shoot runs that version anymore, it can be removed.

That is a behaviour change.

Reply:
resolving this for now, feel free to open again if something is unclear.